doc/developer/arrangements.md

```diff
@@ -166,6 +166,37 @@ In -> <Other agg 1 branch > -> Concat -> CollationInA -> CollationOutA -> Out
    < Other agg 2 branch > ------>
 ```
 
+### Aggregations on distinct values
+
+For non-append-only inputs, hierarchical aggregations on distinct values
```

Comment:
I think this is currently true for all hierarchical aggregates, but its not clear to me that it would be true for any aggregation that we would want to plan hierarchically. I can't think of a good example offhand, but maybe it would be nice to call out that hierarchical aggregations being able to ignore the distinct bit is mostly a quirk of the fact that we only have min/max as hierarchical aggregates, and (perhaps not) a general property of aggregations that correspond to instances of semigroups.